### PR TITLE
Promote Larger Results via Sidecar Logs to Beta

### PR DESCRIPTION
**Pull request by chitrangpatel**
This PR promotes Larger Results via Sidecar Logs to Beta. All items as per Issue https://github.com/tektoncd/pipeline/issues/6133 have been completed. It was released in `v0.43` so has been in an alpha state for a while now.

Fixes: https://github.com/tektoncd/pipeline/issues/6133

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Promote Larger Results via Sidecar Logs to Beta
```
/kind feature

### CODE REVIEW

File: config/config-feature-flags.yaml [hunk 1]
```diff
@@ -108,7 +108,7 @@ data:
   # Setting this flag will determine how Tekton pipelines will handle extracting results from the task.
   # Acceptable values are "termination-message" or "sidecar-logs".
   # "sidecar-logs" is an experimental feature and thus should still be considered
```
Comment (jerop) [inline config/config-feature-flags.yaml:110]: with this promotion to beta, the feature is no longer experimental, please update 

```suggestion
  # "sidecar-logs" is
```

Comment (chitrangpatel) [reply config/config-feature-flags.yaml]: Fixed!

**Review by jerop — APPROVED** [review]
🎉

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**